src/apps/MPI_HALOEXCHANGE.cpp

```diff
@@ -0,0 +1,123 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-23, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "MPI_HALOEXCHANGE.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#if defined(RAJA_PERFSUITE_ENABLE_MPI)
+
+#include <array>
+
+namespace rajaperf
+{
+namespace apps
+{
+
+MPI_HALOEXCHANGE::MPI_HALOEXCHANGE(const RunParams& params)
+  : HALOEXCHANGE_base(rajaperf::Apps_MPI_HALOEXCHANGE, params)
+{
+  m_mpi_size = params.getMPISize();
+  m_my_mpi_rank = params.getMPIRank();
+  m_mpi_dims = params.getMPI3DDivision();
+
+  // TODO: Figure out how to count MPI data movement in BytesPerRep
```

Comment:
I'm not entirely sure how to count the amount of data accessed when MPI is involved.
My first instinct is to count the send as reading all the pack data and the recv as writing to all the unpack data.

Reply:
Is this for the manual count you have for all the kernels?  In that case I'd say that's reasonable.

Reply:
I now count sending as a read of the data and receiving a write of the data.